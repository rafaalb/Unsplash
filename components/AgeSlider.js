import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import Typography from '@material-ui/core/Typography';
import Slider from '@material-ui/lab/Slider';
import Button from '@material-ui/core/Button';

const styles = theme => ({
  root: {
    width: 300,
  },
  slider: {
    padding: '22px 0px',
  },
  button: {
    margin: theme.spacing.unit,
  },
  error: {
    color: 'red',
    fontSize: '1rem'
  }
});

class AgeSlider extends React.Component {
  static defaultProps = {
    classes: {},
    onFilter: () => {}
  };

  state = {
     min: 0,
     max: 100,
     error: null
  };

  onClick = () => {
    const { min, max } = this.state;
    if (min < max) {
      this.props.onFilter(min * 5, max * 5);
    } else {
      this.setState({ error: 'Check your values again' });
    }
  }
  render() {
    const { classes } = this.props;
    const { min, max, error } = this.state;

    return (
      <div className={classes.root}>
        <Typography id="label">Min age: {Math.ceil(min * 5)}</Typography>
        <Slider
          classes={{ container: classes.slider }}
          value={min}
          aria-labelledby="Min Age"
          onChange={(e, val) => this.setState({ min: val, error: null })}
        />
        <Typography id="label">Max age: {Math.ceil(max * 5)}</Typography>
        <Slider
          classes={{ container: classes.slider }}
          value={max}
          aria-labelledby="Max Age"
          onChange={(e, val) => this.setState({ max: val, error: null })}
        />
        {error && <p className={classes.error}>{error}</p>}
        <Button
          variant="outlined"
          color="primary"
          className={classes.button}
          onClick={this.onClick}
        >
          Filter
        </Button>
      </div>
    );
  }
}

AgeSlider.propTypes = {
  classes: PropTypes.object.isRequired,
  onFilter: PropTypes.func,
};

export default withStyles(styles)(AgeSlider);
export { AgeSlider };
