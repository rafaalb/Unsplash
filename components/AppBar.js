import React, { Component } from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import IconButton from '@material-ui/core/IconButton';
import Badge from '@material-ui/core/Badge';
import MailIcon from '@material-ui/icons/Mail';
import NotificationsIcon from '@material-ui/icons/Notifications';
import { withStyles } from '@material-ui/core/styles';
import MenuIcon from '@material-ui/icons/Menu';
import ChevronLeftIcon from '@material-ui/icons/ChevronLeft';
import ChevronRightIcon from '@material-ui/icons/ChevronRight';
import { searchItems } from '../redux/actions'
import { connect } from 'react-redux';
import styled from 'styled-components';
import { drawerWidth } from './Main'

const logo = 'https://upload.wikimedia.org/wikipedia/commons/thumb/4/4a/Unsplash_wordmark_logo.svg/2000px-Unsplash_wordmark_logo.svg.png';
const profileImage = 'https://www.rafaelpina.com/static/profile-f0b81bf77c315c437f6e0955dfcca663.jpg';

const Logo = styled.div`
  @media screen and (max-width: 500px) {
    display: none;
  }
`;

const styles = theme => ({
  appBar: {
    backgroundColor: 'white',
    transition: theme.transitions.create(['margin', 'width'], {
      easing: theme.transitions.easing.sharp,
      duration: theme.transitions.duration.leavingScreen,
    }),
  },
  appBarShift: {
    width: `calc(100% - ${drawerWidth}px)`,
    marginLeft: drawerWidth,
    transition: theme.transitions.create(['margin', 'width'], {
      easing: theme.transitions.easing.easeOut,
      duration: theme.transitions.duration.enteringScreen,
    }),
  },
  grow: {
    flexGrow: 1,
  },
  menuButton: {
    marginLeft: -12,
    marginRight: 20,
    [theme.breakpoints.down('xs')]: {
      marginRight: 0,
    },
  },
  title: {
    transform: 'skew(-5deg) rotate(-4deg)'
  },
  imageProfile: {
    borderRadius: 15,
    height: 30,
    width: 'auto',
    marginLeft: 10,
    border: '1px solid white',
    cursor: 'pointer',
    '&:hover': {
      opacity: 0.7
    },
  },
  inputInput: {
    paddingTop: theme.spacing.unit,
    paddingRight: theme.spacing.unit,
    paddingBottom: theme.spacing.unit,
    paddingLeft: theme.spacing.unit * 10,
    transition: theme.transitions.create('width'),
    width: '100%',
    fontSize: 18,
    [theme.breakpoints.up('sm')]: {
      width: 180,
      paddingLeft: 60,
      '&:focus': {
        width: 250,
      },
    },
    [theme.breakpoints.down('xs')]: {
      paddingLeft: 50,
    },
  },
});

class SearchAppBar extends Component {
  constructor(props) {
    super(props);
    this.state = {
      searchText: ''
    }
  }
  _onEnterPress = (ev) => {
    const { dispatch } = this.props;
    if (ev.key === 'Enter') {
      ev.preventDefault();
      dispatch(searchItems(this.state.searchText));
    }
  }
  handleDrawerClose = () => {
    this.props.handleDrawerClose()
  }
  handleDrawerOpen = () => {
    this.props.handleDrawerOpen()
  }
  render() {
    const { classes, isOpen, onOpen, theme } = this.props;
    return (
      <AppBar
        position="fixed"
        className={classNames(classes.appBar, {
          [classes.appBarShift]: isOpen,
        })}
      >
        <Toolbar>
          {!isOpen && 
            <IconButton
              onClick={onOpen}
              className={classes.menuButton}
              color="inherit"
              aria-label="Open drawer"
            >
              <MenuIcon />
            </IconButton>
          }
          {isOpen &&
            <IconButton onClick={this.handleDrawerClose}>
              {theme.direction === 'ltr' ? <ChevronLeftIcon /> : <ChevronRightIcon />}
            </IconButton>
          }
          <Logo>
            <img
              src={logo}
              alt="logo-image"
              height={25}
              className={classes.title}
            />
          </Logo>
          <div className={classes.grow} />
          <img
            src={profileImage}
            alt="profile-image"
            className={classes.imageProfile}
          />
        </Toolbar>
      </AppBar>
    )
  }
}

SearchAppBar.propTypes = {
  classes: PropTypes.object.isRequired,
};

const Search = withStyles(styles, { withTheme: true })(SearchAppBar);
export default connect()(Search);