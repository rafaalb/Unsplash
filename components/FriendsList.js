import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import List from '@material-ui/core/List';
import Avatar from '@material-ui/core/Avatar';
import Typography from '@material-ui/core/Typography';
import Chip from '@material-ui/core/Chip';
import styled from 'styled-components';
import _ from 'lodash';

const styles = theme => ({
  root: {
    width: '100%',
    maxWidth: 360,
    display: 'flex',
    justifyContent: 'flex-end',
    alignItems: 'flex-end',
    backgroundColor: theme.palette.background.paper,
  },
  secondaryText: {
    color: theme.palette.primary.main,
    padding: 0
  },
  chip: {
    margin: theme.spacing.unit,
  },
});


const StyledPrimary = styled(Typography)`
  && {
    color: ${props => props.theme.blue};
    font-weight: 600;
    font-size: 1.3rem;
  }
`;

class FriendsList extends Component {
  static defaultProps = {
    classes: {}
  }
  renderFriends = () => {
    const { item, classes } = this.props;
    return item.friends.map((friend, i) => (
      <Chip
        avatar={<Avatar alt="Natacha" src={item.thumbnail} />}
        label={friend}
        className={classes.chip}
        key={`${friend}_${i}`}
      />
    ))
  }
  render() {
    const { classes, item } = this.props;
    return (
      <List className={classes.root}>
        <div className={classes.listText}>
          <StyledPrimary>
            {!_.size(item.friends) && `No `}
            Friends
          </StyledPrimary>
          {this.renderFriends()}
        </div>
      </List>
    );
  }
}

FriendsList.propTypes = {
  classes: PropTypes.object.isRequired,
};


export default withStyles(styles)(FriendsList);
export { FriendsList };