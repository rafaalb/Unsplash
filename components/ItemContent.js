import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import List from '@material-ui/core/List';
import ListItem from '@material-ui/core/ListItem';
import ListItemText from '@material-ui/core/ListItemText';
import Avatar from '@material-ui/core/Avatar';
import ImageIcon from '@material-ui/icons/Image';
import FaceIcon from '@material-ui/icons/Face';
import WorkIcon from '@material-ui/icons/Work';
import Typography from '@material-ui/core/Typography';
import styled from 'styled-components';
import StringByCommas from './../lib/StringByCommas';

const styles = theme => ({
  root: {
    width: '100%',
    maxWidth: 360,
    backgroundColor: theme.palette.background.paper,
  },
  secondaryText: {
    color: theme.palette.primary.main,
    padding: 0
  }
});

const StyledText = styled(Typography)`
  && {
    color: ${props => props.theme.black};
    padding: 0;
    font-size: 1rem;
    font-weight: 400;
    line-height: 1;
  }
`;

const StyledPrimary = styled(Typography)`
  && {
    color: ${props => props.theme.blue};
    font-weight: 600;
    font-size: 1.3rem;
    @media screen and (max-width: 550px) {
      font-size: 1rem;
    }
  }
`;

class ItemContainer extends Component {
  render() {
    const { classes, item } = this.props;
    return (
      <List className={classes.root}>
        <ListItem>
          <Avatar>
            <ImageIcon />
          </Avatar>
          <ListItemText
            primary={<StyledPrimary>Weight / Height</StyledPrimary>}
            secondary={
              <StyledText>
                {`${item.weight} x ${item.height}`}
              </StyledText>
            }
          />
        </ListItem>
        <ListItem>
          <Avatar>
            <FaceIcon />
          </Avatar>
          <ListItemText
            primary={<StyledPrimary>Hair Color</StyledPrimary>}
            secondary={<StyledText>{item.hair_color}</StyledText>}
          />
        </ListItem>
        <ListItem>
          <Avatar>
            <WorkIcon />
          </Avatar>
          <ListItemText
            primary={<StyledPrimary>Professions</StyledPrimary>}
            secondary={
              <StyledText>
                {StringByCommas(item.professions)}
              </StyledText>
            }
            classes={classes.listText}
          />
        </ListItem>
      </List>
    );
  }
}

ItemContainer.propTypes = {
  classes: PropTypes.object.isRequired,
};


export default withStyles(styles)(ItemContainer);