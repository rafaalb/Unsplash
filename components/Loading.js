import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import CircularProgress from '@material-ui/core/CircularProgress';
import styled from 'styled-components';

const Center = styled.div`
  text-align: center;
`;

const styles = theme => ({
  progress: {
    margin: theme.spacing.unit * 2,
  },
});

function CircularIndeterminate(props) {
  const { classes } = props;
  return (
    <Center>
      <CircularProgress className={classes.progress} />
    </Center>
  );
}

CircularIndeterminate.propTypes = {
  classes: PropTypes.object.isRequired,
};

export default withStyles(styles)(CircularIndeterminate);
