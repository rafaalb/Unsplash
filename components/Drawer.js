import React, { Component } from 'react'
import classNames from 'classnames';
import { withStyles } from '@material-ui/core/styles';
import DrawerUI from '@material-ui/core/Drawer';
import IconButton from '@material-ui/core/IconButton';
import CssBaseline from '@material-ui/core/CssBaseline';
import Divider from '@material-ui/core/Divider';
import List from '@material-ui/core/List';
import ListItem from '@material-ui/core/ListItem';
import ListItemIcon from '@material-ui/core/ListItemIcon';
import ListItemText from '@material-ui/core/ListItemText';
import FilterIcon from '@material-ui/icons/FilterList';
import ChevronLeftIcon from '@material-ui/icons/ChevronLeft';
import ChevronRightIcon from '@material-ui/icons/ChevronRight';
import SortByAlphaIcon from '@material-ui/icons/SortByAlpha';
import ExpansionPanel from '@material-ui/core/ExpansionPanel';
import ExpansionPanelDetails from '@material-ui/core/ExpansionPanelDetails';
import ExpansionPanelSummary from '@material-ui/core/ExpansionPanelSummary';
import ExpandMoreIcon from '@material-ui/icons/ExpandMore';
import AgeSlider from './AgeSlider';
import SortIcon from '@material-ui/icons/Sort';
import SearchAppBar from './SearchAppBar';
import { sortByName, filterByAge } from './../redux/actions';
import { connect } from 'react-redux';

const drawerWidth = 280;

const styles = theme => ({
  root: {
    display: 'flex',
  },
  appBar: {
    transition: theme.transitions.create(['margin', 'width'], {
      easing: theme.transitions.easing.sharp,
      duration: theme.transitions.duration.leavingScreen,
    }),
  },
  appBarShift: {
    width: `calc(100% - ${drawerWidth}px)`,
    marginLeft: drawerWidth,
    transition: theme.transitions.create(['margin', 'width'], {
      easing: theme.transitions.easing.easeOut,
      duration: theme.transitions.duration.enteringScreen,
    }),
  },
  menuButton: {
    marginLeft: 12,
    marginRight: 20,
  },
  hide: {
    display: 'none',
  },
  drawer: {
    width: drawerWidth,
    flexShrink: 0,
  },
  drawerPaper: {
    width: drawerWidth,
  },
  drawerHeader: {
    display: 'flex',
    alignItems: 'center',
    padding: '0 8px',
    ...theme.mixins.toolbar,
    justifyContent: 'flex-end',
  },
  content: {
    flexGrow: 1,
    marginTop: 50,
    padding: theme.spacing.unit * 3,
    transition: theme.transitions.create('margin', {
      easing: theme.transitions.easing.sharp,
      duration: theme.transitions.duration.leavingScreen,
    }),
    marginLeft: -drawerWidth,
  },
  contentShift: {
    transition: theme.transitions.create('margin', {
      easing: theme.transitions.easing.easeOut,
      duration: theme.transitions.duration.enteringScreen,
    }),
    marginLeft: 0,
  },
});

class Drawer extends Component {
  constructor(props) {
    super(props);
    this.state = {
      open: false,
      expandedAge: false
    }
  }
  handleDrawerOpen = () => {
    this.setState({ open: true });
  };

  handleDrawerClose = () => {
    this.setState({ open: false });
  };
  onClickSort = (property, order) => {
    const { dispatch } = this.props;
    dispatch(sortByName(property, order));
  }
  onClickFilter = (min, max) => {
    const { dispatch } = this.props;
    dispatch(filterByAge(min, max));
  }
  render() {
    const { classes, theme } = this.props;
    const { open, expandedAge } = this.state;

    return (
      <div className={classes.root}>
        <CssBaseline />
        <SearchAppBar
          isOpen={open}
          onOpen={() => this.setState({ open: !this.state.open })}
        />
        <DrawerUI
          className={classes.drawer}
          variant="persistent"
          anchor="left"
          open={open}
          classes={{
            paper: classes.drawerPaper,
          }}
        >
          <div className={classes.drawerHeader}>
            <IconButton onClick={this.handleDrawerClose}>
              {theme.direction === 'ltr' ? <ChevronLeftIcon /> : <ChevronRightIcon />}
            </IconButton>
          </div>
          <Divider />
          <List>
            <ListItem button disabled>
              <ListItemText primary={'Sort options'} />
            </ListItem>
            <ListItem button onClick={() => this.onClickSort('name', 'asc')}>
              <ListItemIcon><SortByAlphaIcon /></ListItemIcon>
              <ListItemText primary={'Sort by name ASC'} />
            </ListItem>
            <ListItem button onClick={() => this.onClickSort('name', 'desc')}>
              <ListItemIcon><SortIcon /></ListItemIcon>
              <ListItemText primary={'Sort by name DESC'} />
            </ListItem>
            <ListItem button onClick={() => this.onClickSort('age', 'asc')}>
              <ListItemIcon><SortByAlphaIcon /></ListItemIcon>
              <ListItemText primary={'Sort by age'} />
            </ListItem>
          </List>
          <Divider />
          <List>
            <ExpansionPanel
              expanded={expandedAge}
              onChange={() => this.setState({ expandedAge: !this.state.expandedAge })}
              style={{ boxShadow: 'none' }}
            >
              <ExpansionPanelSummary style={{ paddingLeft: 0 }} expandIcon={<ExpandMoreIcon />}>
                <ListItem>
                  <ListItemIcon>
                    <FilterIcon />
                  </ListItemIcon>
                  <ListItemText primary={'Filter by age'} />
                </ListItem>
              </ExpansionPanelSummary>
              <ExpansionPanelDetails>
                <AgeSlider onFilter={this.onClickFilter} />
              </ExpansionPanelDetails>
            </ExpansionPanel>
            <Divider />
          </List>
        </DrawerUI>
        <main
          className={classNames(classes.content, {
            [classes.contentShift]: open,
          })}
        >
          <div className={classes.drawerHeader} />
          {this.props.children}
        </main>
      </div>
    );
  }
}

const StyledDrawer = withStyles(styles, { withTheme: true })(Drawer);
export default connect()(StyledDrawer);
export { drawerWidth };