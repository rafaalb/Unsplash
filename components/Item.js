import React, { Component } from 'react'
import PropTypes from 'prop-types'
import Title from './styles/Title';
import ItemStyles from './styles/ItemStyles';
import Link from 'next/link';
import Age from './styles/Age';
import ItemContent from './ItemContent';
import FriendsList from './FriendsList';
import styled from 'styled-components';

const ContentContainer = styled.div`
  position: relative;
  display: flex;
  @media screen and (max-width: 880px) {
    position: unset;
  }
`;
export default class Item extends Component {
  static propTypes = {
    item: PropTypes.object.isRequired
  }

  render() {
    const { item } = this.props;
    return (
      <ItemStyles>
        {item.thumbnail &&  <img src={item.thumbnail} />}
        <Title>
          <Link>
            <a>{item.name}</a>
          </Link>
        </Title>
        <ContentContainer>
          <ItemContent item={item} />
          <FriendsList item={item} />
          <Age>
            {`${item.age} years old`}
          </Age>
        </ContentContainer>
      </ItemStyles>
    )
  }
}
