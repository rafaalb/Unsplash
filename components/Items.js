import React, { Component } from 'react'
import styled from 'styled-components';
import { connect } from 'react-redux'; 
import Item from './Item';
import { fetchItems, addItems, fetchUsers } from '../redux/actions';
import _size from 'lodash/size';
import Loading from './Loading';
import Button from '@material-ui/core/Button';
import SearchBar from './SearchBar'

const Center = styled.div`
  text-align: center;
`;

const ItemsList = styled.div`
  display: grid;
  grid-gap: 40px;
  margin: 0 auto;
  grid-template-columns: 1fr 1fr;
  @media screen and (max-width: 500px) {
    grid-template-columns: 1fr;
  }
`;

const UserInfo = styled.div`
  text-align: left;
  margin-bottom: 30px;
  h3 {
    color: white;
  }
`;

const options = [
  'Carlos',
  'Mary',
  'Miguel',
  'Emily',
  'Rafael',
  'Andres'
]

const Suggest = ({ classes, onEnterPress, search }) => {
  return (
    <>
      <h4>
        Start searching for amazing photos
      </h4>
      <SearchBar
        onEnterPress={onEnterPress}
        toggleError={() => {}}
      />
      <h6>or you can check any of these</h6>
      {options.map(option =>
        <Button
          variant="outlined"
          color="primary"
          style={{ margin: 5 }}
          onClick={() => search(option)}
        >
            {option}
        </Button>
      )}
    </>
  )
}

class Items extends Component {
  componentDidMount() {
    this.props.fetchItems();
  }
  _renderItems = () => {
    const { selected } = this.props.users
    if (!_size(selected.photos)) return;
    return selected.photos.map((item, i) => (
      <Item item={item} key={`${item.id}_${i}`} />
    ));
  }
  _onEnterPress = (payload) => {
    return this.props.fetchUsers(payload);
  }
  render() {
    const { items, users } = this.props;
    if (items.searching) return <Loading />;
    if (users.firstLoad) return (
      <Suggest onEnterPress={this._onEnterPress} search={this.props.fetchUsers} />
    )
    return (
      <Center>
        <UserInfo>
          <h3>{this.props.users.selected.name}</h3>
          <em>{this.props.users.selected.bio}</em>
        </UserInfo>
        <ItemsList>
          {this._renderItems()}
        </ItemsList>
      </Center>
    );
  }
}
const mapDispatchToProps = (dispatch) => {
  return {
    fetchUsers: (payload) => dispatch(fetchUsers(payload)),
    fetchItems: (payload) => dispatch(fetchItems(payload))
  }
}

export default connect(state => state, mapDispatchToProps)(Items);
