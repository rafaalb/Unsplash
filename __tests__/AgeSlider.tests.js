import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import { AgeSlider } from '../components/AgeSlider';

describe('', () => {
  it('renders', () => {
    shallow(<AgeSlider />)
  });
  it('renders and matches the snapshot', () => {
    const wrapper = shallow(
      <AgeSlider />
    );
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
  it('should call filter on button click', () => {
    const onFilter = jest.fn();
    const wrapper = shallow(
      <AgeSlider onFilter={onFilter} />
    );
    const button = wrapper.find('WithStyles(Button)');
    button.simulate('click');
    expect(onFilter).toHaveBeenCalled();
  })
});