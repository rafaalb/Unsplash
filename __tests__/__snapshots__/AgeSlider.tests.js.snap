// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders and matches the snapshot 1`] = `
<div>
  <WithStyles(Typography)
    id="label"
  >
    Min age: 
    0
  </WithStyles(Typography)>
  <WithStyles(Slider)
    aria-labelledby="Min Age"
    classes={
      Object {
        "container": undefined,
      }
    }
    onChange={[Function]}
    value={0}
  />
  <WithStyles(Typography)
    id="label"
  >
    Max age: 
    500
  </WithStyles(Typography)>
  <WithStyles(Slider)
    aria-labelledby="Max Age"
    classes={
      Object {
        "container": undefined,
      }
    }
    onChange={[Function]}
    value={100}
  />
  <WithStyles(Button)
    color="primary"
    onClick={[Function]}
    variant="outlined"
  >
    Filter
  </WithStyles(Button)>
</div>
`;
